supabase/migrations/20251229000001_sso_auto_join_fixes.sql

```diff
@@ -0,0 +1,393 @@
+    RETURN 'NO_RIGHTS';
+  END IF;
+
+  -- Delete domain
+  DELETE FROM public.org_domains WHERE id = p_domain_id;
+
+  RETURN 'OK';
+END;
+$$;
+
+-- Update update_org_domain_settings to accept optional user_id
+CREATE OR REPLACE FUNCTION public.update_org_domain_settings(
+  p_domain_id uuid,
+  p_auto_join_enabled boolean DEFAULT NULL,
+  p_auto_join_role public.user_min_right DEFAULT NULL,
+  p_user_id uuid DEFAULT NULL
+)
+RETURNS text
+LANGUAGE plpgsql
+SECURITY DEFINER
+SET search_path = ''
+AS $$
+DECLARE
+  v_org_id uuid;
+  v_effective_user_id uuid;
+BEGIN
+  v_effective_user_id := COALESCE(p_user_id, auth.uid());
+
+  -- Get org_id for the domain
+  SELECT org_id INTO v_org_id FROM public.org_domains WHERE id = p_domain_id;
+
+  IF v_org_id IS NULL THEN
+    RETURN 'DOMAIN_NOT_FOUND';
+  END IF;
+
+  -- Check if user has super_admin rights
+  IF NOT public.check_min_rights('super_admin'::public.user_min_right, v_effective_user_id, v_org_id, NULL::varchar, NULL::bigint) THEN
+    RETURN 'NO_RIGHTS';
+  END IF;
+
+  -- Update settings
+  UPDATE public.org_domains
+  SET
+    auto_join_enabled = COALESCE(p_auto_join_enabled, auto_join_enabled),
+    auto_join_role = COALESCE(p_auto_join_role, auto_join_role),
+    updated_at = now()
+  WHERE id = p_domain_id;
+
+  RETURN 'OK';
+END;
+$$;
+
+-- Update add_org_domain to accept optional user_id
+CREATE OR REPLACE FUNCTION public.add_org_domain(
+  p_org_id uuid,
+  p_domain varchar,
+  p_user_id uuid DEFAULT NULL
+)
+RETURNS TABLE (
+  id uuid,
+  verification_token varchar,
+  error_code text
+)
+LANGUAGE plpgsql
+SECURITY DEFINER
+SET search_path = ''
+AS $$
+DECLARE
+  v_token varchar;
+  v_id uuid;
+  v_effective_user_id uuid;
+BEGIN
+  v_effective_user_id := COALESCE(p_user_id, auth.uid());
+
+  -- Check if user has super_admin rights
+  IF NOT public.check_min_rights('super_admin'::public.user_min_right, v_effective_user_id, p_org_id, NULL::varchar, NULL::bigint) THEN
+    RETURN QUERY SELECT NULL::uuid, NULL::varchar, 'NO_RIGHTS'::text;
+    RETURN;
+  END IF;
+
+  -- Check if org is Enterprise
+  IF NOT public.is_enterprise_org(p_org_id) THEN
+    RETURN QUERY SELECT NULL::uuid, NULL::varchar, 'REQUIRES_ENTERPRISE'::text;
+    RETURN;
+  END IF;
+
+  -- Check if domain is already claimed
+  IF EXISTS (SELECT 1 FROM public.org_domains WHERE domain = lower(p_domain)) THEN
+    RETURN QUERY SELECT NULL::uuid, NULL::varchar, 'DOMAIN_ALREADY_CLAIMED'::text;
+    RETURN;
+  END IF;
+
+  -- Generate verification token
+  v_token := encode(gen_random_bytes(32), 'hex');
+
+  -- Insert domain
+  INSERT INTO public.org_domains (org_id, domain, verification_token)
+  VALUES (p_org_id, lower(p_domain), v_token)
+  RETURNING org_domains.id INTO v_id;
+
+  RETURN QUERY SELECT v_id, v_token, NULL::text;
+END;
+$$;
+
+-- Update upsert_org_sso_provider to accept optional user_id
```

Comment:
I think we can drop the old function

```suggestion
-- Update upsert_org_sso_provider to accept optional user_id
-- First drop the old 6-parameter version to avoid overloading conflict
DROP FUNCTION IF EXISTS public.upsert_org_sso_provider(uuid, uuid, varchar, varchar, text, boolean);

```